declarations/index.d.ts

```diff
@@ -1,22 +1,107 @@
 /// <reference types="node" />
 import { EventEmitter } from 'events';
-export interface IAbstractIO {
-    isReady: boolean;
-    readonly MODES: {
-        INPUT: number;
-        OUTPUT: number;
-        ANALOG: number;
-        PWM: number;
-        SERVO: number;
-    };
+export declare enum Mode {
+    INPUT = 0,
+    OUTPUT = 1,
+    ANALOG = 2,
+    PWM = 3,
+    SERVO = 4,
+    UNKOWN = 99
+}
+export declare enum Value {
+    HIGH = 1,
+    LOW = 0
+}
+export interface IPinConfiguration {
+    supportedModes: Mode[];
+    mode: Mode;
+    value: Value;
```

Comment:
Although it's not documented in the IO Plugins specification I think the values that value can have are pin-mode specific. For digital pins value can be 0 or 1 but for analog pins I think a range of values are allowed. I'm not 100% sure about this though.

Reply:
That is correct. The firmata protocol allows for any bit-depth on analog pins so it could be any unsigned integer. Currently, we only see 8-bit or 10-bit but the protocol allows for more.

If we have a capability query response from the board, we know the possible range but I imagine that information comes too late in the process to be of value here.

Reply:
Do we know for certain that, regardless of bit depth, it will always be a number? In TypeScript, we can either set a type as a `number`, meaning a JavaScript 64-bit floating point number, or a set of specific values (either through an enum or a type union, e.g. `0 | 1  | 2`). It sounds like we should switch the type to `number`, but I want to be certain it can't be, say, a string.

Reply:
Not strictly documented, but looking at the firmata.js code there is no expected case where the value will be anything other than Number. I see that if an anlogWrite is sent with something other than a Number for value, that value will be stored locally as whatever type was passed but the message sent to the board will be bit-shifted, which means coerced to Number if possible. Otherwise it's zero. In this sense, the remote value may not match the local value which is a bug IMHO and you are about to fix it.

TLDR:
Make it ```Number```